### PR TITLE
add Google Analytics tracking context

### PR DESCRIPTION
**Pull request by justincorrigible**

### CODE REVIEW

File: .env.schema [hunk 1]
```diff
@@ -20,5 +20,8 @@ NEXT_PUBLIC_ARRANGER_MANIFEST_COLUMNS=field, "field", 'field'
 ######### DMS
 NEXT_PUBLIC_SSO_PROVIDERS=
 
+# ######## Tracking
+NEXT_PUBLIC_GOOGLE_ANALYTICS_ID=
```
Comment (justincorrigible) [inline .env.schema:24]: add value to the template for .env

File: global/config.ts [hunk 2]
```diff
@@ -36,26 +36,29 @@ type AppConfig = {
...
     NEXT_PUBLIC_ARRANGER_INDEX: publicConfig.NEXT_PUBLIC_ARRANGER_INDEX || '',
     NEXT_PUBLIC_ARRANGER_ADMIN_UI: publicConfig.NEXT_PUBLIC_ARRANGER_ADMIN_UI,
-    NEXT_PUBLIC_ARRANGER_API_URL: publicConfig.NEXT_PUBLIC_ARRANGER_API_URL || '',
     NEXT_PUBLIC_ARRANGER_MANIFEST_COLUMNS: publicConfig.NEXT_PUBLIC_ARRANGER_MANIFEST_COLUMNS || '',
     NEXT_PUBLIC_BASE_PATH: publicConfig.NEXT_PUBLIC_BASE_PATH || '/',
     NEXT_PUBLIC_ADMIN_EMAIL: publicConfig.NEXT_PUBLIC_ADMIN_EMAIL,
     NEXT_PUBLIC_LAB_NAME: publicConfig.NEXT_PUBLIC_LAB_NAME || 'Data Management System',
     NEXT_PUBLIC_LOGO_FILENAME: publicConfig.NEXT_PUBLIC_LOGO_FILENAME,
     NEXT_PUBLIC_SSO_PROVIDERS: publicConfig.NEXT_PUBLIC_SSO_PROVIDERS || '',
+    NEXT_PUBLIC_GOOGLE_ANALYTICS_ID: publicConfig.NEXT_PUBLIC_GOOGLE_ANALYTICS_ID || '',
```
Comment (justincorrigible) [inline global/config.ts:62]: then plug it into NextJS

File: package.json [hunk 3]
```diff
@@ -27,6 +27,7 @@
     "react": "16.13.1",
     "react-dom": "16.13.1",
     "react-grid-system": "^7.1.1",
+    "react-ga": "^3.3.0",
```
Comment (justincorrigible) [inline package.json:30]: package used to extend the GA api

File: global/hooks/useTrackingContext/types.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+import { event } from 'react-ga';
+
+export type LogEventFunctionType = typeof event;
+
+export type TrackingContextType = {
+  addTracker: (trackerId: string, trackerName: string) => void;
+  logEvent: LogEventFunctionType;
+  removeTracker: (trackerName: string) => void;
+};
```
Comment (justincorrigible) [inline global/hooks/useTrackingContext/types.ts:30]: this is the context being provided:
- a function to connect to a tracker (aka GA-UID)
- a function to trigger a custom event listener in the tracker
- and a way to stop an existing tracker

File: global/hooks/useTrackingContext/types.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+export type TrackingStateType = {
+  isInitialized: boolean;
+  hasUser: boolean;
+  trackers: string[];
+};
```
Comment (justincorrigible) [inline global/hooks/useTrackingContext/types.ts:36]: just the custom hook's internal state

File: components/Root.tsx [hunk 6]
```diff
@@ -63,7 +64,9 @@ const Root = ({
       <Head />
       <AuthProvider sessionToken={sessionToken} initialUser={initialUser}>
         <PageContext.Provider value={pageContext}>
-          <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
+          <TrackingProvider>
+            <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
+          </TrackingProvider>
```
Comment (justincorrigible) [inline components/Root.tsx:69]: we provide the context here

**Review by justincorrigible — COMMENTED** [review]
couple details to reason with the changes 🧠

**Review by anncatton — APPROVED** [review]

**Pull request merged**